base/compiler/abstractlattice.jl

```diff
@@ -131,6 +131,23 @@ remaining mostly associative and commutative.
 """
 function tmerge end
 
+"""
+    tmerge_field(𝕃::AbstractLattice, a, b) -> nothing or lattice element
+
+Compute a lattice join of elements `a` and `b` over the lattice `𝕃`,
```

Comment:
This is still not quite the lattice that tmerge refers to. It is a complexity reduction join, like tmerge. But it deviates from tmerge because it is a join of invariant parameters instead of a covariant join, so the requirements are even a bit stricter